src/Network/Network/Common/NetworkResourceManagerProfile.cs

```diff
@@ -89,6 +89,37 @@ public override string ProfileName
             }
         }
 
+        public class VirtualRouterConverter : ITypeConverter<MNM.VirtualRouter, CNM.PSVirtualRouter>
+        {
+            public CNM.PSVirtualRouter Convert(MNM.VirtualRouter source, CNM.PSVirtualRouter destination, ResolutionContext context)
+            {
+                var router = new CNM.PSVirtualRouter()
+                {
+                    Id = source.Id,
+                    Name = source.Name,
+                    Etag = source.Etag,
+                    Location = source.Location,
+                    Type = source.Type,
+                    ProvisioningState = source.ProvisioningState,
+                    VirtualRouterAsn = (uint) source.VirtualRouterAsn
```

Comment:
VirtualRouterAsn is nullable in SDK model, right? This could cause an exception during casting

Reply:
VirtualRouterAsn is long in sdk. I dont think long can be null